presto-docs/src/main/sphinx/connector/pinot.rst

```diff
@@ -0,0 +1,105 @@
+                                                              Pinot will return hostnames and not ip addresses.
+``pinot.segments-per-split``                       No         The number of segments processed in a split. Setting this higher will reduce
+                                                              the number of requests made to Pinot which makes sense for a smaller Pinot cluster.
+``pinot.request-timeout``                          No         The timeout for Pinot requests. Increasing this can reduce timeouts if dns
+                                                              resolution is slow.
+================================================== ========== ===================================================================================
+
+Querying Pinot Tables
+-------------------------
+
+The schema for each Pinot catalog is ``default``. To find all tables you can run::
+
+    SHOW TABLES FROM pinot.default;
+
+To see a list of columns in the ``flight_status`` table the following can be run:
+
+    DESCRIBE pinot.default.flight_status;
+    SHOW COLUMNS FROM pinot.default.flight_status;
+
+Tables can be queried as they are in other connectors, leveraging filter and limit pushdown:
+
+    SELECT foo FROM pinot_table WHERE bar = 3 AND baz IN ('ONE', 'TWO', 'THREE') LIMIT 25000;
+
+Dynamic Tables
+--------------
+
+To leverage Pinot's fast aggregation, a Pinot query written in PQL can be used as the table name.
+Filters and limits in the outer query will be pushed down to Pinot.
+Some examples:
+
+    SELECT *
+    FROM pinot.default."SELECT MAX(col1), COUNT(col2) FROM pinot_table GROUP BY col3, col4"
+    WHERE col3 IN ('FOO', 'BAR') AND col4 > 50
+    LIMIT 30000
+
+The filter in the outer presto query will be pushed down into the Pinot query via Presto's applyFilter(). These queries are routed to the broker and
+should not return huge amounts of data as broker queries currently return a single response with all the results. This
+is more suited to aggregate queries.
+
+Limits are pushed into the "dynamic" pinot query via Presto's applyLimit(). The above query would yield the following Pinot PQL query:
+
+    SELECT MAX(col1), COUNT(col2)
+    FROM pinot_table
+    WHERE col3 IN('FOO', 'BAR') and col4 > 50
+    TOP 30000
+
+
+If you are returning a larger dataset you can issue a normal Presto query which will get routed to the Pinot servers which
+store the Pinot segments. Filters and Limits are pushed down to Pinot for regular queries as well.
+
+Data types
+----------
+
+Pinot supports the following primitive types which currently not allow null values:
+
+==========================   ============
+Pinot                        Presto
+==========================   ============
+``INT``                      ``INTEGER``
+``LONG``                     ``BIGINT``
+``FLOAT``                    ``REAL``
+``DOUBLE``                   ``DOUBLE``
+``STRING``                   ``VARCHAR``
+``INT_ARRAY``                ``VARCHAR``
```

Comment:
Why are these being mapped to `varchar` instead of, for instance, `array(integer)`?